python/amici/ode_export.py

```diff
@@ -3114,7 +3027,7 @@ def _get_function_body(self,
                             f'reinitialization_state_idxs.cend(), {index}) != '
                             'reinitialization_state_idxs.cend())',
                             f'    {function}[{index}] = '
-                            f'{_print_with_exception(formula)};'
+                            f'{self.model._code_printer.doprint(formula)};'
```

Comment:
wouldn't it make more sense to have `_code_printer` as member of the exporter?

Reply:
Fully agreed. The problem is that ODEModel also generates code (which I think should be changed, but maybe rather separately). Having it in ODEModel for now allows using it also in ODEExporter, but not vice versa.